chore: one round of updates

huitseeker opened the pull request:
- Updates the mysten-infra pointer, and updates to gets rid of pesky no-longer existing features in tracing and crossbeam crates,
- removes a no longer needed hack to not import bzip2 in rocksdb

huitseeker committed:
chore: one round of updates

asonnino reviewed (approved):
Cool, let's land it now before I make a big rebase for https://github.com/MystenLabs/narwhal/pull/447

huitseeker committed:
fix: remove unneeded dependency

huitseeker committed:
fix: turn off windows CI temporarily

We run out of disk on GH-hosted runners, and will reinstate those builds
as soon as we have self-hosted runners.
